vendor/github.com/mongodb/mongo-tools-common/bsonutil/bsonutil.go

```diff
@@ -324,91 +319,35 @@ func ParseSpecialKeys(special interface{}) (interface{}, error) {
 				} else if len(kind) != 1 {
 					return nil, errors.New("expected single byte (as hexadecimal string) for $type field")
 				}
-				binary.Kind = kind[0]
+				binary.Subtype = kind[0]
 
 			default:
 				return nil, errors.New("expected $type field to have string value")
 			}
 			return binary, nil
 		}
-
-		if jsonValue, ok := doc["$ref"]; ok {
```

Comment:
the go driver has no type associated with a dbRef (and neither does bson, for that matter), so I just removed all the parsing for it. It should go through the parser as a document, which is basically what it is anyways.